react-carousel/my-app/src/Carousel.tsx

```diff
@@ -0,0 +1,44 @@
+import './App';
+import { ImgCard } from './ImgCard';
+import { Prev } from './Prev';
+import { Next } from './Next';
+import { Dots } from './Dots';
+import { useEffect, useState } from 'react';
+
+type ImgProp = {
+  src: string;
+  alt: string;
+};
+
+type Prop = {
+  details: ImgProp[];
+};
+
+export function Carousel({ details }: Prop) {
+  const [cycle, setCycle] = useState(0);
+  function handlePrev(): void {
+    setCycle((cycle - 1 + details.length) % details.length);
+  }
+
+  function handleNext(): void {
+    setCycle((cycle + 1) % details.length), [details.length, cycle];
+  }
+
+  useEffect(() => {
+    const intervalId = setInterval(handleNext, 1000);
+    return () => clearInterval(intervalId);
+  }, [handleNext]);
```

Comment:
This should give you a warning on handleNext. That function needs to be wrapped in useCallback.